src/interfaces/swig/shogun.i

```diff
@@ -246,6 +276,22 @@ namespace shogun
 		return mat;
 	}
 #endif // SWIGJAVA
+
+#ifdef SWIGR
+	template <typename T, typename X = typename std::enable_if_t<std::is_same<SGVector<typename extract_value_type<T>::value_type>, T>::value>>
+	void put_vector_scalar_dispatcher(const std::string& name, T vector)
+	{
+		if (Tag<T> tag_vec(name); vector.size() > 1 || $self->has(tag_vec))
+		{
+			$self->put(tag_vec, vector);
+		}
+		else
+		{
+			auto value = vector[0];
+			PUT_SCALAR_DISPATCHER(X, name, value)
+		}
+	}
+#endif // SWIGR
```

Comment:
@vigsterkr We can drop all these hacks (R, Java, and Python "put" type trying), once smart visitors are in right? I think we should also allow casting between SGVector and SGMatrix since octave has a problem with that currently.